pkg/bff/collaborators.go

```diff
@@ -93,3 +93,49 @@ func (s *Server) ReplaceCollaborator(c *gin.Context) {
 
 	c.JSON(http.StatusOK, collaborator)
 }
+
+// DeleteCollaborator deletes the collaborator in the request from the user's
+// organization. The user must have the update:collaborators permission.
+// Note: This does not return an error if the collaborator does not exist on the
+// organization and instead returns a 200 OK response.
+func (s *Server) DeleteCollaborator(c *gin.Context) {
+	var (
+		err          error
+		collaborator *models.Collaborator
+		org          *models.Organization
+	)
+
+	// Fetch the organization from the claims
+	// NOTE: This method handles the error logging and response
+	if org, err = s.OrganizationFromClaims(c); err != nil {
+		return
+	}
+
+	// Unmarshal the collaborator from the DELETE request
+	collaborator = &models.Collaborator{}
+	if err = c.ShouldBind(collaborator); err != nil {
+		log.Warn().Err(err).Msg("could not bind request")
+		c.JSON(http.StatusBadRequest, api.ErrorResponse(err))
+		return
+	}
+
+	// Make sure we can delete the collaborator by key
+	key := collaborator.Key()
+	if key == "" {
+		log.Warn().Err(err).Msg("missing key on collaborator")
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("invalid collaborator in request"))
+		return
+	}
+
+	// Delete the collaborator from the organization
+	delete(org.Collaborators, key)
```

Comment:
The current code does not return an error if the collaborator doesn't exist, but I suppose we could check the map and return a 404 to ensure that the backend and frontend are on the same page?

Reply:
@masskoder any thoughts about the status code you'd like returned?

Reply:
@pdeziel @bbengfort  it would be great if we could have 200 (if okay ). 204 , if the collaborators ID is not found or invalid.

Reply:
@masskoder 2XX codes are meant to be used for successful statuses, in my view an invalid collaborator key/ID is definitely an error. If we are changing the PUT and DELETE endpoints to accept IDs (e.g., `DELETE /v1/collaborators/:collabid`) perhaps the endpoint should return 400 for an invalid collabid and a 404 if the collaborator is missing. Does that make sense?